src/main/scala/mesosphere/marathon/state/PathId.scala

```diff
@@ -88,4 +88,37 @@ object PathId {
     def toPath: PathId = PathId(stringPath)
     def toRootPath: PathId = PathId(stringPath).canonicalPath()
   }
-}
+
+  /**
+    * This regular expression is used to validate each path segment of an ID.
+    *
+    * If you change this, please also change "pathType" in AppDefinition.json and
+    * notify the maintainers of the DCOS CLI.
+    */
+  private[this] val ID_PATH_SEGMENT_PATTERN =
+    "^(([a-z0-9]|[a-z0-9][a-z0-9\\-]*[a-z0-9])\\.)*([a-z0-9]|[a-z0-9][a-z0-9\\-]*[a-z0-9])|(\\.|\\.\\.)$".r
+
+  import com.wix.accord.dsl._
+  implicit val pathIdValidator = validator[PathId] { pathId =>
+    pathId.path.each should matchRegex(ID_PATH_SEGMENT_PATTERN.pattern)
+    // pathId is validChild(pathId.parent)
+  }
+
+  import com.wix.accord._
+  import scala.util.Try
+  def validChild(parent: PathId): Validator[PathId] = {
+    new Validator[PathId] {
+      override def apply(child: PathId): Result = {
+        Try(child.canonicalPath(parent)) match {
```

Comment:
`parent` should always be an absolute path (an exception is thrown if not). Handling that case here (and also catching all exceptions) I would try to avoid.